pimd/pim_nb_config.c

```diff
@@ -1914,11 +1923,8 @@ int lib_interface_pim_bfd_destroy(struct nb_cb_destroy_args *args)
 	case NB_EV_APPLY:
 		ifp = nb_running_get_entry(args->dnode->parent, NULL, true);
 		pim_ifp = ifp->info;
-		if (pim_ifp->bfd_info) {
-			pim_bfd_reg_dereg_all_nbr(ifp,
-					ZEBRA_BFD_DEST_DEREGISTER);
-			bfd_info_free(&(pim_ifp->bfd_info));
-		}
+		pim_bfd_reg_dereg_all_nbr(ifp, false);
+		pim_ifp->bfd_config.enabled = false;
```

Comment:
If we remove `enable` argument from `pim_bfd_reg_dereg_all_nbr` as I suggested earlier, then `pim_ifp->bfd_config.enabled` should be set to false before the call.

Reply:
Fixed